### PR TITLE
Add Iso8601Highlighter

### PR DESCRIPTION
**Pull request by tomers**
## Type of changes

- [ ] Bug fix
- [X] New feature
- [ ] Documentation / docstrings
- [ ] Tests
- [ ] Other

## Checklist

- [X] I've run the latest [black](https://github.com/psf/black) with default args on new code.
- [X] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [X] I've added tests for new code.
- [X] I accept that @willmcgugan may be pedantic in the code review.

## Description

As suggesterd by @willmcgugan in #2295, I've added a date-time highlighter.

### CODE REVIEW

File: rich/highlighter.py [hunk 1]
```diff
@@ -140,6 +140,64 @@ def highlight(self, text: Text) -> None:
                 break
 
 
+class Iso8601Highlighter(RegexHighlighter):
```
Comment (willmcgugan) [inline rich/highlighter.py:143]: I think camel case would require it to be `ISO8601Highlighter`

**Review by willmcgugan — CHANGES_REQUESTED** [review]
This is great. Could you add something to the docs (probably in the highlighter section).

**Comment by tomers** [comment]
@willmcgugan, I've force-pushed without realizing that you submitted to the remote branch. Can you re-introduce your changes? Sorry for that.

**Comment by tomers** [comment]
@willmcgugan, what else need to be done in order for this to be merged?

**Comment by willmcgugan** [comment]
Thanks

**Pull request merged**